src/cls.ts

```diff
@@ -80,30 +82,34 @@ export class TraceCLS implements CLS<RootContext> {
   readonly rootSpanStackOffset: number;
 
   constructor(private readonly logger: Logger, config: TraceCLSConfig) {
-    const useAH = config.mechanism === 'async-hooks' && asyncHooksAvailable;
-    if (useAH) {
-      this.CLSClass = AsyncHooksCLS;
-      this.rootSpanStackOffset = 4;
-      this.logger.info(
-          'TraceCLS#constructor: Created [async-hooks] CLS instance.');
-    } else {
-      if (config.mechanism !== 'async-listener') {
-        if (config.mechanism === 'async-hooks') {
+    switch (config.mechanism) {
+      case 'async-hooks':
+        if (asyncHooksAvailable) {
+          this.CLSClass = AsyncHooksCLS;
+          this.rootSpanStackOffset = 4;
+          break;
+        } else {
           this.logger.error(
               'TraceCLS#constructor: [async-hooks]-based context',
               `propagation is not available in Node ${process.version}.`);
-        } else {
-          this.logger.error(
-              'TraceCLS#constructor: The specified CLS mechanism',
-              `[${config.mechanism}] was not recognized.`);
+          throw new Error(`CLS mechanism [${config.mechanism}] is invalid.`);
```

Comment:
nit: the logged error message seem a lot more appropriate. Can you just use that? Also, do we need to log given that we are throwing?

Reply:
follow-on: nit: reverse the if so that the throw happens early. Remove the indentation of the else block.